File: Python/basic_comunication/smartphone.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class SmartphoneNode(Node):
    def __init__(self):
        super().__init__("smartphone")
        self.subscriber_ = self.create_subscription(
            String, "robot_news", self.callback_robot_news, 10) #1- Tipo da msg #2 - nome do tópico #3- função callback #4- queue size
        self.get_logger().info("Smartphone has been started.")

    def callback_robot_news(self, msg: String): # callback que será chamado toda vez que o subscriber receber a mensagem
        self.get_logger().info(msg.data) # printa a prórria mensagem recebida


def main(args = None):
    rclpy.init(args=args)
    node = SmartphoneNode()
    rclpy.spin(node)
    rclpy.shutdown()

    

if __name__ == '__main__':
    main()

File: C++/basic_comunication/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"// incluindo a interface de mensagem

// Opcional: comolar "using namespace std::placeholder" para não precisar escrever tudo na linha 10 

class SmartphoneNode : public rclcpp::Node{
    public:
        SmartphoneNode() : Node("smartphone"){ 
           subscriber_ = this->create_subscription<example_interfaces::msg::String>(
            "robot_news", 10, std::bind(&SmartphoneNode::callbackRobotNews, this, std::placeholders::_1));
            RCLCPP_INFO(this->get_logger(), "Smartphone has been started.");
        }
    private:
       void callbackRobotNews(const example_interfaces::msg::String::SharedPtr msg){
        RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
       }

       rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;

};

int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SmartphoneNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
